staging/src/k8s.io/apimachinery/pkg/runtime/types_proto.go

```diff
@@ -24,46 +24,66 @@ type ProtobufMarshaller interface {
 	MarshalTo(data []byte) (int, error)
 }
 
+type ProtobufReverseMarshaller interface {
+	LahsramOt(data []byte) (int, error)
+}
+
 // NestedMarshalTo allows a caller to avoid extra allocations during serialization of an Unknown
-// that will contain an object that implements ProtobufMarshaller.
+// that will contain an object that implements ProtobufReverseMarshaller.
```

Comment:
Technically you've implemented both, if that's intentional I'd say `ProtobufMarshaller or ProtobufReverseMarshaller`